testing/matrix-sdk-integration-testing/src/tests/sliding_sync/room.rs

```diff
@@ -1126,6 +1128,207 @@ async fn test_room_preview() -> Result<()> {
     Ok(())
 }
 
+#[tokio::test]
+async fn test_edit_unpaginated_item_after_resetting_timeline() -> Result<()> {
```

Comment:
Can you extract out those two integration tests, and post a new PR with those, as they seem unrelated to the feature at hand, please?

Reply:
well they are actually also a way to test the feature, since they can address both the issue we currently have and an integration test implementation of editing an item that is not in the timeline

Reply:
If an integration test doesn't trigger a failure you observe with new functions introduced, let's try to keep the PR lean and remove it from this PR and open another instead; it'll make this PR easier to review, and investigating why the test doesn't pass easier, without blocking one with the other.